functions/utils/serve.ts

```diff
@@ -0,0 +1,72 @@
+import { ConnInfo, serve } from "https://deno.land/std@0.167.0/http/server.ts";
+import { MyceliumClient } from "./graphql-client.ts";
+import { makeResp } from "./make-resp.ts";
+
+interface EdgeFunctionHandlerParams<RuntimeDataType = unknown> {
+  myceliumClient: MyceliumClient;
+  request: Request;
+  connInfo: ConnInfo;
+  runtimeData: RuntimeDataType;
+}
+
+interface EdgeFunctionHandlerOptions {
+  apiUrl?: string;
+}
+
+type EdgeFunctionHandler<RuntimeDataType> = (
+  params: EdgeFunctionHandlerParams<RuntimeDataType>,
+) => Response | Promise<Response>;
+
+export function serveTruffleEdgeFunction<RuntimeDataType = unknown>(
+  handler: EdgeFunctionHandler<RuntimeDataType>,
+  options?: EdgeFunctionHandlerOptions,
+) {
+  return serve(async (request, connInfo) => {
+    const accessToken = request.headers.get("x-access-token");
+    const orgId = request.headers.get("x-org-id");
+
```

Comment:
do we want to add any validation that the request is coming from our api or should 3rd parties be able to call this function?

Reply:
I'll add a todo

Reply:
Update: https://github.com/trufflehq/truffle-packages/pull/72/commits/6256d4dbdd866e24e4605ca469ae4e238455e019